lib/cfg-block-generator.c

```diff
@@ -38,7 +38,14 @@ gboolean
 cfg_block_generator_generate(CfgBlockGenerator *self, GlobalConfig *cfg, CfgArgs *args, GString *result,
                              const gchar *reference)
 {
-  return self->generate(self, cfg, args, result, reference);
+  gchar block_name[1024];
+  cfg_block_generator_format_name(self, block_name, sizeof(block_name)/sizeof(block_name[0]));
```

Comment:
Just a minor: [G_N_ELEMENTS](https://developer.gnome.org/glib/stable/glib-Standard-Macros.html#G-N-ELEMENTS:CAPS) could be used.